fibheap/fibheap_test.go

```diff
@@ -0,0 +1,427 @@
+	7229910558.195713, 8122228442.770859, 8985474333.11443}
+
+var NumberSequence3 = [...]float64{6015943293.071386, -3878285748.0708866, 8674121166.062424,
+	-1528465047.6118088, 7584260716.494843, -373958476.80486107, -6367787695.054295,
+	6813992306.719868, 5986097626.907181, 9011134545.052086, 7123644338.268343,
+	2646164210.08445, 4407427446.995375, -888196668.2563229, 7973918726.985172,
+	-6529216482.09644, 6079069259.51853, -8415952427.784341, -6859960084.757652,
+	-502409126.89040375}
+
+var NumberSequence4 = [...]float64{9241165993.258648, -9423768405.578083, 3280085607.6687145,
+	-5253703037.682413, 3858507441.2785892, 9896256282.896187, -9439606732.236805,
+	3082628799.5320206, 9453124863.59945, 9928066165.458393, 1135071669.4712334,
+	6380353457.986282, 8329064041.853199, 2382910730.445751, -8478491750.445316,
+	9607469190.690144, 5417691217.440792, -9698248424.421888, -3933774735.280322,
+	-5984555343.381466}
+
+var NumberSequenceMerged3And4Sorted = [...]float64{-9698248424.421888, -9439606732.236805,
+	-9423768405.578083, -8478491750.445316, -8415952427.784341, -6859960084.757652,
+	-6529216482.09644, -6367787695.054295, -5984555343.381466, -5253703037.682413,
+	-3933774735.280322, -3878285748.0708866, -1528465047.6118088, -888196668.2563229,
+	-502409126.89040375, -373958476.80486107, 1135071669.4712334, 2382910730.445751,
+	2646164210.08445, 3082628799.5320206, 3280085607.6687145, 3858507441.2785892,
+	4407427446.995375, 5417691217.440792, 5986097626.907181, 6015943293.071386,
+	6079069259.51853, 6380353457.986282, 6813992306.719868, 7123644338.268343,
+	7584260716.494843, 7973918726.985172, 8329064041.853199, 8674121166.062424,
+	9011134545.052086, 9241165993.258648, 9453124863.59945, 9607469190.690144,
+	9896256282.896187, 9928066165.458393}
+
+func TestEnqueueDequeueMin(t *testing.T) {
+	heap := NewFloatFibHeap()
+	for i := 0; i < len(NumberSequence1); i++ {
+		heap.Enqueue(NumberSequence1[i])
+	}
+
+	var min *Entry
+	var err error
+	for heap.Size() > 0 {
+		min, err = heap.DequeueMin()
+		assert.NoError(t, err)
+		if heap.Size() == 199 {
+			assert.Equal(t, Seq1FirstMinimum, min.Priority)
+		}
+		if heap.Size() == 197 {
+			assert.Equal(t, Seq1ThirdMinimum, min.Priority)
+		}
+		if heap.Size() == 195 {
+			assert.Equal(t, Seq1FifthMinimum, min.Priority)
+		}
+		if heap.Size() == 0 {
+			assert.Equal(t, Seq1LastMinimum, min.Priority)
+		}
+	}
+}
+
+func TestFibHeap_Enqueue_Min(t *testing.T) {
+	heap := NewFloatFibHeap()
+	for i := 0; i < len(NumberSequence1); i++ {
+		heap.Enqueue(NumberSequence1[i])
+	}
+
+	min, err := heap.Min()
+	assert.NoError(t, err)
+	assert.Equal(t, Seq1FirstMinimum, min.Priority)
+}
+
+func TestFibHeap_Min_EmptyHeap(t *testing.T) {
+	heap := NewFloatFibHeap()
+
+	heap.Enqueue(0)
+	heap.DequeueMin()
+
+	// Heap should be empty at this point
+
+	min, err := heap.Min()
+
+	assert.EqualError(t, err, "Trying to get minimum element of empty heap")
```

Comment:
It's not that clumsy and people do do that, but.. I suppose what you propose is more sophisticated and elegant.